zebra-consensus/src/transaction.rs

```diff
@@ -169,6 +169,12 @@ where
                 check::coinbase_tx_no_prevout_joinsplit_spend(&tx)?;
             }
 
+            // [Canopy onward]: `vpub_old` MUST be zero.
+            // https://zips.z.cash/protocol/protocol.pdf#joinsplitdesc
+            if req.upgrade(network) >= NetworkUpgrade::Canopy {
+                check::disabled_sprout_pool(&tx)?;
+            }
```

Comment:
@jvff should we try to move this sprout-specific check inside `verify_sprout_shielded_data`?

I don't mind either way.

Reply:
I think that might make sense, because V5 transactions don't need this check since they don't have Sprout shielded data.

Reply:
@oxarbitrage can you please make this change?

Reply:
Sorry, this is not clear for me. Do you guys mean something as the following into `verify_sprout_shielded_data`
```
// [Canopy onward]: `vpub_old` MUST be zero.
// https://zips.z.cash/protocol/protocol.pdf#joinsplitdesc
checks.push(check::disabled_add_to_sprout_pool(&tx, req.height(), network).boxed());
```

Note that we don't have any of the 3 parameters currently available in `verify_sprout_shielded_data` so i guess we have to change the `disabled_add_to_sprout_pool` signature and/or pass arguments into `verify_sprout_shielded_data`.
But i think i am not following the overall idea.

Please clarify, thanks.

Reply:
Yes, that's basically it.

Here are some more suggestions for changes to `verify_sprout_shielded_data`:
* `joinsplit_data` contains the joinsplit balancing amounts
* you'll need to pass `height` and `network` as arguments
* if we change `verify_sprout_shielded_data` to return `Result<AsyncChecks, TransactionError>`, we can return a failure straight away

Reply:
I'll leave the final review here to @jvff.

Reply:
This is now https://github.com/ZcashFoundation/zebra/issues/2435